src/lib/internal_tracing/internal_tracing.ml

```diff
@@ -202,5 +219,77 @@ module For_logger = struct
   let processor = Logger.Processor.create (module Processor) ()
 end
 
+module For_itn_logger = struct
+  let last_block_id = ref no_block
+
+  let last_call_id = ref 0
+
+  (* Used by ITN logging handler. Unlike the above, this produces raw messages
+     instead of JSON output.
+
+     IMPORTANT: this must replicate the same logic as [For_logger.Processor.process] *)
+  let post_process_message ~timestamp ~message ~metadata =
+    let handling_current_block_id_change =
+      handling_current_block_id_change ~last_block_id ~make:(fun block_id ->
+          ( timestamp
+          , "@control"
+          , [ ("current_block", `String (Block_id.to_string block_id)) ] ) )
+    in
+    let handling_current_call_id_change =
+      handling_current_call_id_change ~last_call_id ~make:(fun tag call_id ->
+          ( timestamp
+          , "@control"
+          , [ ("current_call_id", `Int call_id)
+            ; ("current_call_tag", `String tag)
+            ] ) )
+    in
+    if is_enabled () then
+      let log_messages : (Time.t * string * (string * Yojson.Safe.t) list) list
+          =
+        let open For_logger.Processor.Event in
+        let assoc = Yojson.Safe.Util.to_assoc in
+        let metadata = String.Map.of_alist_reduce ~f:(fun a _b -> a) metadata in
+        match message with
+        | "@metadata" ->
+            handling_current_call_id_change
+            @@ handling_current_block_id_change
+                 [ (timestamp, "@control", assoc @@ checkpoint_metadata metadata)
+                 ]
+        | "@block_metadata" ->
```

Comment:
Would it be useful to use an OCaml class + objects to enforce that the same protocol is used here, as in  `For_logger` (or a record containing functions?

Reply:
Ok, let me see what I can do about that.

As a side-note, what I think would be the best option here (as far as exposing internal tracing goes) is to handle this logic at the processor level (instead of the logging calls), so that there is no need to duplicate any of the logic. So instead, what would be exposed (and sent between processes) is whatever output is produced by the json-lines processor (still using the same LogID logic to be able to filter out older logs when performing queries).

But I am not sure if there are plans to use this mechanism for something else, and also there was a GraphQL schema already defined and documented, so I didn't want to introduce such incompatible change.

Reply:
Ok, left a local network running for a while, just checked the output to compare the files produced by the node with the logs obtained from the remote logs API and they look fine.